Keep current back link when navigating through the topics folder side panel

rtibbles opened the pull request:
## Summary
* Fixes error in link generation in folder side navigation panel and ensures that the current back link is preserved during navigation - to be identical to clicking on a folder card in the main navigation panel

## References
Fixes [#11432](https://github.com/learningequality/kolibri/issues/11432)

## Reviewer guidance
Click on a folder in the left side panel.
Press the X to close out the immersive modal, ensure that it does close.

Do the same when exploring a remote library and ensure navigation occurs correctly.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [x] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

rtibbles committed:
Keep current back link when navigating through the topics folder side panel.

marcellamaki reviewed (commented):
code change makes sense to me - seems like the better function to use (and luckily past Richard already did the hard part of making it!). I'll let @radinamatic confirm that it's covering all of the scenarios she reported.

radinamatic commented:
Not yet, apparently... I tested the DEB asset in both Chrome & Firefox on Ubuntu, and the issue persists 🤔 

https://github.com/learningequality/kolibri/assets/1457929/4db7ce1f-2eaa-4f5c-b8e6-2b3eb18cb4fe

Only one error noted in the Chrome console, not sure if it's relevant.

<img width="419" alt="2023-10-20_19-56-30" src="https://github.com/learningequality/kolibri/assets/1457929/6640b741-331b-4b94-927e-1b6661e1426b">

rtibbles committed:
Rename variable to make it clear it is always a folder.
Add backlink utility for adding the previous back link.
Use it in for Folder links from the TopicsContentPage.

rtibbles commented:
This scenario should now also be covered!

pcenov reviewed (approved):
Works fine now!

radinamatic reviewed (approved):
Manual QA confirms the fix, good to go! 💯 🎉 :shipit:

marcellamaki reviewed (approved):
Additional code changes look good to me. The renaming for folder variable is also helpful - thanks!
